packages/@react-aria/overlays/src/useModal.tsx

```diff
@@ -123,8 +122,7 @@ interface OverlayContainerProps extends ModalProviderProps {
  * be accessible at once.
  */
 export function OverlayContainer(props: OverlayContainerProps): React.ReactPortal {
-  let isSSR = useIsSSR();
-  let {portalContainer = isSSR ? null : document.body, ...rest} = props;
+  let {portalContainer = typeof document !== 'undefined' ? document.body : null, ...rest} = props;
```

Comment:
This won't work because during hydration (the first render on the client side), the output must match what was rendered on the server, or React will warn. The server would return `null` below because document is undefined on the server. So during hydration, we must also do this on the first render, and update afterward. That's what `useIsSSR` does.

Could you explain why this caused an issue for you?